Assets/Scripts/Features/DayNightCycle/DayNightCycle.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DayNightCycle : MonoBehaviour
+{
+    [SerializeField]
+    private Transform _lightSourceTransform;
+    private bool _rotating;
+
+    private void Start()
+    {
+        PlayTool.Instance.OnSecondPassedEvent.AddListener(RotateLightSource);
+       
+    }
+
+    private void RotateLightSource()
+    {
+        Debug.Log("rotating");
+        if (!_rotating)
+            StartCoroutine(Rotate(new Vector3(1, 0, 0), 1));
```

Comment:
The bool could be prevented with using an "instance" of a coroutine which contains it, and is halted when a new coroutine is engaged.
Other then that, very well made. ! =)